### PR TITLE
clean up publishing

### PR DESCRIPTION
**Pull request by smurthas**
This makes the way two things are treated with regard to publishing app consistent:

**1. handle and name are forced to be <github-login>-<repo-name), e.g. smurthas-linkpic.** - This means that all locker apps with have a unique global handle and it will consistent with npm's unique id (name).

**2. description is moved into the repository field.** - now _all_ person-facing data is contained and managed within the repository field.

Also, updates to dashboard to support it and an update to the github repos synclet to conform to this method.

finally, a little tweak to lservicemanager to allow absolute paths to be upserted. It will still populate the srcdir field with a path relative to lockerDir because the passed path is just used to go find the package.json file.

(PS, can now pass a arg into Ops/upsert.js with a dir, saves some cd-ing)

### CODE REVIEW

**Comment by quartzjer** [comment]
THANK YOU!

**Comment by smurthas** [comment]
BTW, this shouldn't be merged yet, it might cause problems with old github repos data + map. Might need a simple migration, needs to be tested against prod data (or real, data from yesterday-ish).

File: Common/node/lservicemanager.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ exports.providers = function(types) {
 exports.mapUpsert = function (file) {
     var js;
     try {
-        js = JSON.parse(fs.readFileSync(path.join(lconfig.lockerDir,file), 'utf8'));
```
Comment (quartzjer) [inline Common/node/lservicemanager.js:127]: There's places lower in this function that depend on file being relative not absolute, and I'm not sure the other srcdir logic is safe elsewhere either (just eyeballed some as was a bit uncomfortable).

Maybe I'll add in something to calculate the relative local path from an absolute one if any is passed in with a leading / to be totally safe :)

**Comment by temas** [comment]
I disagree with the description being in the repository field.  We're using the npm variant commonjs package format and we should be good citizens.  Plus it feels absolutely silly to not use a correctly existing field.  This should be true for any field that is replicated in the spec.

We'll come around, eventually, to needing to cleanup the repository usage to actually be a more correct array and then dependencies too.  It can't all be shoved in one spot that we found works.

**Comment by smurthas** [comment]
I totally agree on being good citizens of NPM, so I'd like to use the top level description field, but the reality is it's very hard to remember what goes where. What I think we could do is to simply force those two fields to be consistent, such that someone can enter a description inside of the repository and have it propagated to the high level field (or vice versa).

Does work?

**Comment by quartzjer** [comment]
@smurthas on the migration possible need, if anything I think it might just need to have the repos synclet re-run to re-upsert any possible apps (and it always normalizes the local package.json in the process)?  A global post-startup migration that looks for the github dir and does a /synclets/github/run?id=repos would work if so?

**Comment by quartzjer** [comment]
btw, I'm +1 now on this whenever y'all are ready :)

**Comment by smurthas** [comment]
I tested this on my prod data and it does a few weird things without a migration. Should be simple, but I'm going to try to track down what looks likes a bug due to the propagation of id, handle, and provider (inconsistent in one place).

A migration looks like it will still be needed to safely merge this.

**Comment by quartzjer** [comment]
And deleting any pkgs in registry that were bogus too :)

**Comment by smurthas** [comment]
K, that migrations should deal with things. Straight forward conversion to the new format (including changing dirs to lowercase, almost missed that with my case insensitive file system!).

Could a few others try this, preferably with some prod data if you've got it.

**Comment by othiym23** [comment]
+1 on merging to master. After @smurthas and I had a highly entertaining debugging session, I was able to run the migration correctly against Me/ and my sample viewer was migrated correctly into my locker, with the map looking clean.

**Pull request merged**